extensions/pkg/controller/worker/machines.go

```diff
@@ -170,31 +173,42 @@ func WorkerPoolHashV1(pool extensionsv1alpha1.WorkerPool, cluster *extensionscon
 	return utils.ComputeSHA256Hex([]byte(result))[:5], nil
 }
 
-// WorkerPoolHashV2 returns a hash value for a given pool and additional data.
-func WorkerPoolHashV2(pool extensionsv1alpha1.WorkerPool, cluster *extensionscontroller.Cluster, additionalData ...string) (string, error) {
-	var data []string
+// WorkerPoolHashV2 returns a hash value for a given nodeAgentSecretName and additional data.
+func WorkerPoolHashV2(nodeAgentSecretName string, additionalData ...string) (string, error) {
+	data := []string{nodeAgentSecretName}
+
+	data = append(data, additionalData...)
+
+	var result string
+	for _, v := range data {
+		result += utils.ComputeSHA256Hex([]byte(v))
+	}
+
+	return utils.ComputeSHA256Hex([]byte(result))[:5], nil
+}
+
+// WorkerPoolHashInPlace return hash value for a give worker pool with update strategy in-place.
```

Comment:
// WorkerPoolHashInPlace returns the hash value for a worker pool with an in-place update strategy.